### PR TITLE
Monitior Section storage and use joins_allowed switch

### PR DESCRIPTION
**Pull request by Yoga07**

### CODE REVIEW

File: src/network.rs [hunk 1]
```diff
@@ -117,6 +117,15 @@ impl Network {
         // Ok(())
     }
 
+    pub async fn set_node_join(&mut self, joins_allowed: bool) -> Result<()> {
```
Comment (lionel1704) [inline src/network.rs:120]: Perhaps a rename will do good here?

File: src/node/elder_duties/key_section/mod.rs [hunk 2]
```diff
@@ -86,6 +94,13 @@ impl<R: CryptoRng + Rng> KeySection<R> {
         self.transfers.catchup_with_replicas().await
     }
 
+    pub async fn set_node_join_flag(&mut self, joins_allowed: bool) -> Outcome<NodeOperation> {
```
Comment (lionel1704) [inline src/node/elder_duties/key_section/mod.rs:97]: Here too. Perhaps rename this to `set_joins_allowed` ?

**Comment by lionel1704** [comment]
Left a few small comments. Once those are addressed and this is rebased on top of the Dev branch we can merge this.

**Comment by Yoga07** [comment]
Closing in favour of https://github.com/maidsafe/sn_node/pull/1172

**Pull request closed without merging**